### PR TITLE
Add mutations to manage authorization keys

### LINKED ISSUE #3080: Mutations to mange authorization keys
Add mutations to add/remove authorization keys.

### PR DESCRIPTION
**Pull request by maarcingebala**
Fixes #3080 

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.

### CODE REVIEW

File: saleor/graphql/api.py [hunk 1]
```diff
@@ -271,6 +272,9 @@ def resolve_address_validator(self, info, input):
 
 
 class Mutations(graphene.ObjectType):
+    authorization_key_add = AuthorizationKeyAdd.Field()
+    authotization_key_delete = AuthorizationKeyDelete.Field()
```
Comment (Pacu2) [inline saleor/graphql/api.py]: typo `authotization` -> `authorization`

File: saleor/graphql/shop/mutations.py [hunk 2]
```diff
@@ -108,3 +108,69 @@ def mutate(cls, root, info, collection):
...
+class AuthorizationKeyAdd(BaseMutation):
+    authorization_key = graphene.Field(
+        AuthorizationKey, description='Newly added authorization key.')
+
+    class Meta:
+        description = 'Adds an authorization key.'
+
+    class Arguments:
+        key_type = AuthorizationKeyType(
+            required=True, description='Type of authorization key to add.')
```
Comment (Pacu2) [inline saleor/graphql/shop/mutations.py]: I think we are missing `an` before the `authorization` (as in the description below)

File: saleor/graphql/shop/mutations.py [hunk 3]
```diff
@@ -108,3 +108,69 @@ def mutate(cls, root, info, collection):
...
+    @permission_required('site.manage_settings')
+    def mutate(cls, root, info, key_type, input):
+        errors = []
+        if site_models.AuthorizationKey.objects.filter(name=key_type).exists():
+            cls.add_error(
+                errors, 'key_type', 'Authorization key already exists.')
+            return AuthorizationKeyAdd(errors=errors)
+
+        site_settings = info.context.site.settings
+        instance = site_models.AuthorizationKey.objects.create(
```
Comment (Pacu2) [inline saleor/graphql/shop/mutations.py]: Do we want to save it at this point, or should we just create an instance, without saving it, then run `cls.clean_instance`, and if there's no errors - save it?

File: saleor/graphql/shop/mutations.py [hunk 4]
```diff
@@ -108,3 +108,69 @@ def mutate(cls, root, info, collection):
...
+        return AuthorizationKeyAdd(authorization_key=instance)
+
+
+class AuthorizationKeyDelete(BaseMutation):
+    authorization_key = graphene.Field(
+        AuthorizationKey, description='Auhtorization key that was deleted.')
+
+    class Arguments:
+        key_type = AuthorizationKeyType(
+            required=True, description='Type of key to delete.')
```
Comment (Pacu2) [inline saleor/graphql/shop/mutations.py]: `Type of a key`

File: saleor/graphql/shop/mutations.py [hunk 5]
```diff
@@ -108,3 +108,69 @@ def mutate(cls, root, info, collection):
...
+
+    class Meta:
+        description = 'Deletes an authorization key.'
+
+    @classmethod
+    @permission_required('site.manage_settings')
+    def mutate(cls, root, info, key_type):
+        errors = []
+        try:
```
Comment (Pacu2) [inline saleor/graphql/shop/mutations.py]: Personally I like `first()` a little bit more than try-except, I find it more readable

File: saleor/graphql/shop/types.py [hunk 6]
```diff
@@ -53,7 +62,10 @@ class Shop(graphene.ObjectType):
         Geolocalization,
         description='Customer\'s geolocalization data.')
     authorization_keys = graphene.List(
-        AuthorizationKey, description='List of configured authorization keys.',
+        AuthorizationKey,
+        description='''List of configured authorization keys. Authorization
+        keys are used to enable thrid party OAuth authorization (currently
```
Comment (Pacu2) [inline saleor/graphql/shop/types.py]: To be precise it's `OAuth2`

Comment (maarcingebala) [reply saleor/graphql/shop/types.py]: Is the concept of `client_id` and `client_secret` fields specific only to OAuth2 and not valid in OAuth1? I meant general concept of OAuth.

File: saleor/graphql/shop/mutations.py (hunk 5 above)
Comment (maarcingebala) [reply saleor/graphql/shop/mutations.py]: `try..except` pattern is also used across our codebase and API as well, both are readable in my opinion.

File: saleor/graphql/shop/types.py (hunk 6 above)
Comment (Pacu2) [reply saleor/graphql/shop/types.py]: As far as I know it's the same concept, matter of fact if we want to say precisely that OAuth2 authorization is supported by us, or that this feature can be used for OAuth in general

Comment (maarcingebala) [reply saleor/graphql/shop/types.py]: `python-social-auth` which we use under the hood provides backends for both OAuth1 and OAuth2 authorizations. All of them could be potentially enabled in Saleor, so I think that generally saying about OAuth here should be fine.

**Pull request merged**